homeassistant/components/transmission.py

```diff
@@ -0,0 +1,171 @@
+            port = config[DOMAIN].get(CONF_PORT)
+
+            api = transmissionrpc.Client(
+                host, port=port, user=username, password=password)
+
+            """Initialize the Transmission data object."""
+            self.data = None
+            self.torrents = None
+            self.available = True
+            self._api = api
+            self.completed_torrents = []
+            self.started_torrents = []
+            self.initTorrentList()
+            self.hass = hass
+
+        except TransmissionError as error:
+            if str(error).find("401: Unauthorized"):
+                _LOGGER.error("Credentials for"
+                              " Transmission client are not valid")
+                return
+
+            _LOGGER.warning(
+                "Unable to connect to Transmission client: %s:%s", host, port)
+            raise PlatformNotReady
+
+    @Throttle(SCAN_INTERVAL)
+    def update(self):
+        """Get the latest data from Transmission instance."""
+        from transmissionrpc.error import TransmissionError
+
+        try:
+            self.data = self._api.session_stats()
+            self.torrents = self._api.get_torrents()
+
+            self.checkCompletedTorrent()
+            self.checkStartedTorrent()
+
+            _LOGGER.info("Torrent Data updated")
+            self.available = True
+        except TransmissionError:
+            self.available = False
+            _LOGGER.error("Unable to connect to Transmission client")
+
+    def initTorrentList(self):
+        self.torrents = self._api.get_torrents()
+        self.completed_torrents = [
+            x.name for x in self.torrents if x.status == "seeding"]
+        self.started_torrents = [
+            x.name for x in self.torrents if x.status == "downloading"]
+
+    def checkCompletedTorrent(self):
+        """Get completed torrent functionality"""
+        actual_torrents = self.torrents
+        actual_completed_torrents = [
+            var.name for var in actual_torrents if var.status == "seeding"]
+
+        tmp_completed_torrents = list(
+            set(actual_completed_torrents).difference(
+                self.completed_torrents))
+        if tmp_completed_torrents:
+            for var in tmp_completed_torrents:
+                _LOGGER.info("Completed Torrent found")
+                self.hass.bus.fire(
+                    'transmission_downloaded_torrent', {
+                        'name': var})
+
+        self.completed_torrents = actual_completed_torrents
+
+    def checkStartedTorrent(self):
+        """Get started torrent functionality"""
+        actual_torrents = self.torrents
+        actual_started_torrents = [
+            var.name for var
+            in actual_torrents if var.status == "downloading"]
+
+        tmp_started_torrents = list(
+            set(actual_started_torrents).difference(
+                self.started_torrents))
+        if tmp_started_torrents:
```

Comment:
This can be removed.

Reply:
Removed how? I need the tmp_started_torrents variables for event triggering

Reply:
We don't need to check if it's non empty before iterating the list.